packages/material-ui/src/internal/SwitchBase.d.ts

```diff
@@ -60,8 +60,10 @@ export interface SwitchBaseProps
    * @param {object} event The event source of the callback.
    * You can pull out the new value by accessing `event.target.value` (string).
    * You can pull out the new checked state by accessing `event.target.checked` (boolean).
+   *
+   * @deprecated This signature `(event, checked) => void` will be deprecated in a future release. Use `(event) => { event.target.checked }` instead.
```

Comment:
Is it semantically correct to use `@deprecated` for changing a parameter in a function? I am not sure I have a better proposal, I am just not sure that this is correct. To me `@deprecated` means that the `onChange` can be replaced with something else. I may be wrong tough..

Reply:
My purpose is to have it in api docs.
![image](https://user-images.githubusercontent.com/18292247/124110272-40ab9880-da92-11eb-878d-075e922d116e.png)

> @deprecated means that the onChange can be replaced with something else

This make sense, some people might think `onChange` is deprecated. I will revert the comment. May be the migration docs is enough.